scylla/tests/integration/statement.rs

```diff
@@ -309,3 +317,754 @@ async fn test_timestamp_generator() {
+        .execute_unpaged(&insert_a_b_c, values_dyn)
+        .await
+        .unwrap();
+
+    let values_box_dyn: Vec<Box<dyn SerializeValue>> = vec![
+        Box::new(1) as Box<dyn SerializeValue>,
+        Box::new(3) as Box<dyn SerializeValue>,
+        Box::new("Box dyn") as Box<dyn SerializeValue>,
+    ];
+    session
+        .execute_unpaged(&insert_a_b_c, values_box_dyn)
+        .await
+        .unwrap();
+
+    let mut all_rows: Vec<(i32, i32, String)> = session
+        .query_unpaged("SELECT a, b, c FROM tab", ())
+        .await
+        .unwrap()
+        .into_rows_result()
+        .unwrap()
+        .rows::<(i32, i32, String)>()
+        .unwrap()
+        .map(|r| r.unwrap())
+        .collect();
+    all_rows.sort();
+    assert_eq!(
+        all_rows,
+        vec![
+            (1i32, 2i32, "&dyn".to_owned()),
+            (1, 3, "Box dyn".to_owned())
+        ]
+    );
+}
+
+const KEY: &str = "test";
+
+/// Initialize a cluster with two tables table and insert some data into one of them.
+/// Returns a session and the keyspace name.
+///
+/// # Example
+/// ```rust
+/// let (session, ks) = initialize_cluster_two_tables().await;
+/// ```
+async fn initialize_cluster_two_tables() -> (Session, String) {
+    setup_tracing();
+    let session = create_new_session_builder().build().await.unwrap();
+
+    let ks = unique_keyspace_name();
+
+    session.ddl(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1}}", ks)).await.unwrap();
+    session
+        .ddl(format!(
+            "CREATE TABLE IF NOT EXISTS {}.test (k text, v int, PRIMARY KEY(k, v))",
+            ks
+        ))
+        .await
+        .unwrap();
+
+    for i in 0..100 {
+        session
+            .query_unpaged(
+                format!("INSERT INTO {}.test (k, v) VALUES (?, ?)", ks),
+                (KEY, i),
+            )
+            .await
+            .unwrap();
+    }
+
+    session
+        .ddl(format!(
+            "CREATE TABLE IF NOT EXISTS {}.test2 (k text primary key, v int)",
+            ks
+        ))
+        .await
+        .unwrap();
+
+    (session, ks)
+}
+
+#[tokio::test]
+async fn should_fail_when_too_many_positional_values_provided() {
+    let (session, ks) = initialize_cluster_two_tables().await;
+    let query = Query::new(format!("SELECT v FROM {}.test WHERE k=?", ks));
+
+    let result = session.query_unpaged(query.clone(), (KEY, 1)).await;
+    match result {
+        Err(ExecutionError::LastAttemptError(RequestAttemptError::SerializationError(e))) => {
+            match e
+                .downcast_ref::<serialize::row::BuiltinTypeCheckError>()
+                .unwrap()
+                .kind
+            {
+                serialize::row::BuiltinTypeCheckErrorKind::WrongColumnCount {
+                    rust_cols: 2,
+                    cql_cols: 1,
+                } => {}
+                _ => panic!("Expected WrongColumnCount error"),
```

Comment:
:wrench: `assert_matches!` prints more information upon no match, so let's use it.

Reply:
```suggestion
            let kind = &e
                .downcast_ref::<serialize::row::BuiltinTypeCheckError>()
                .unwrap();
                .kind;
            assert_matches!(kind, serialize::row::BuiltinTypeCheckErrorKind::WrongColumnCount {
                rust_cols: 2,
                cql_cols: 1,
            });
```